### PR TITLE
segger: system-view: include path fix (+ some minor improvements) - regression

### LINKED ISSUE #61133: segger: system-view: include path priority issue (regression)
**Describe the bug**

The Zephyr-specific Segger SystemView configuration is no longer used as an empty file with the same name was inadvertently added to the external segger module with higher import priority during the last Segger module upgrade a few months ago.

**To Reproduce**

1. Build any application with the following configuration:
CONFIG_TRACING=y
CONFIG_THREAD_NAME=y
CONFIG_USE_SEGGER_RTT=y
CONFIG_SEGGER_SYSTEMVIEW=y
2. Start the program
3. An immediate fault caused by infinite recursion will result

**Expected behavior**

The application should start normally and tracing with Segger SystemView should work.

**Impact**

The Segger SystemView tracing integration became unusable (regression).

**Logs and console output**

n/a

**Environment (please complete the following information):**

any - reproduced on Linux

**Additional context**

Fix exists already - PR will be linked.

### PR DESCRIPTION
**Pull request by ghost**
This fixes a regression in the the Segger SystemView integration: The Zephyr-specific Segger SystemView configuration was not used as the same file was available inside the external segger module with higher import priority since the last Segger module upgrade a few months ago. This made Segger SystemView unusable (regression).

Some other minor improvements in separate commits:
* ~~It is proposed to change the custom RTT locking strategy from mutexes to spinlocks:~~
    * ~~Mutexes caused infinite loops when tracing with Segger SystemView.~~
    * ~~The prior implementation did not support RTT logging and tracing from ISR context.~~
    * ~~Logging and tracing are performance-critical features. Spinlocks are more lightweight and faster than mutexes. AFAICS mutex features like re-entrance or priority inheritance are not required for RTT locking plus yielding due to trace instrumentation should be avoided when tracing IMHO.~~ 
UPDATE: **The include priority problem turned out to be the root cause of the problems described here so the spinlock-part of the change set has been withdrawn. It's remaining minor theoretical merit doesn't justify such a heavy change IMO**, see the discussion in https://github.com/zephyrproject-rtos/segger/pull/19.
* Showcases logging via RTT UART which is now available due to the preceding change.
* Fixes a few bugs in the formatting of the Segger SystemView output.
* Fixes a minor documentation bug in the Segger SystemView documentation.    

This change set has been tested on CC1352R hardware with Segger's JLink.

Fixes: #61133

### CODE REVIEW

**Comment by fgrandel** [comment]
@vaishnavachath @niflostancu @cfriedt @jadonk Tagging you as I believe you have an interest in TI CC13/26xx support. This change set implicitly fixes Segger SystemView for CC13/26xx (+ all other ARM-based TI SoCs) - see this screenshot as an example generated on a CC1352R SensorKit:

segger-systemview

This tool will be valuable to debug TDMA protocols and all other kinds of timing-sensitive stuff on the TI DevKits or BeagleBone. Works concurrently with RTT output (which has also recently been fixed for CC13/26xx).

File: west.yml [hunk 1]
```diff
@@ -304,7 +306,8 @@ manifest:
       path: modules/lib/picolibc
       revision: d07c38ff051386f8e09a143ea0a6c1d6d66dd1d8
     - name: segger
-      revision: 5792675a2470d0f3857de1e77bff57b38c28de3b
+      remote: fgrandel
+      revision: fix/rtt-locking
```
Comment (pdgendt) [inline west.yml:310]: No need to add a remote in `west.yml` you can point to the PR's `HEAD`:

```suggestion
      revision: pull/19/head
```

Comment (fgrandel) [reply west.yml]: @pdgendt Nice. Fixed.

**Comment by fgrandel** [comment]
I believe that the failing compliance check is unrelated to my change.

**Comment by carlescufi** [comment]
@fgrandel please replace revision with a SHA

**Review by carlescufi — DISMISSED** [review]
This looks good to me, thanks!

**Comment by fgrandel** [comment]
> @fgrandel please replace revision with a SHA

@carlescufi Done.

**Review by carlescufi — APPROVED** [review]
re-ack

**Review by nashif — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]

**Pull request merged**